lib/SILGen/SILGenProlog.cpp

```diff
@@ -482,7 +488,16 @@ uint16_t SILGenFunction::emitProlog(ParameterList *paramList,
   auto *genericSig = DC->getGenericSignatureOfContext();
   resultType = resultType->getCanonicalType(genericSig);
 
-  emitIndirectResultParameters(*this, resultType, DC);
+  if (!emitIndirectResultParameters(*this, resultType, DC)) {
```

Comment:
Cleaning this up would be nice in general since we want SILGenProlog and SILGenEpilog to handle emitting functions at different levels of abstraction directly. This will let us avoid wrapping literal closures in dynamically allocated thunks when they're passed generically, and such.